### PR TITLE
adding github action to test plugin installation via homebrew

### LINKED ISSUE #66: Add simple install test for PRs
there should be a GitHub action that tests the install of the plugins, this simple test should catch simple errors like on reported on Slack today

Hello Knative, long time no see. I’m really wanting to check out the new quickstart system but it looks like I might have caught it at a bad time or might be doing something wrong. When I try to install it I get:
❯ brew install knative-sandbox/kn-plugins/quickstart
==> Tapping knative-sandbox/kn-plugins
Cloning into '/usr/local/Homebrew/Library/Taps/knative-sandbox/homebrew-kn-plugins'...
remote: Enumerating objects: 203, done.
remote: Counting objects: 100% (203/203), done.
remote: Compressing objects: 100% (152/152), done.
remote: Total 203 (delta 126), reused 66 (delta 43), pack-reused 0
Receiving objects: 100% (203/203), 50.63 KiB | 4.22 MiB/s, done.
Resolving deltas: 100% (126/126), done.
Error: Invalid formula: /usr/local/Homebrew/Library/Taps/knative-sandbox/homebrew-kn-plugins/event@1.1.rb
No available formula with the name "event@1.1". Did you mean event@1.1?
In formula file: /usr/local/Homebrew/Library/Taps/knative-sandbox/homebrew-kn-plugins/event@1.1.rb
Expected to find class EventAT11, but only found: Event.
Error: Cannot tap knative-sandbox/kn-plugins: invalid syntax in tap!

### PR DESCRIPTION
**Pull request by sbdtu5498**
## adding github action to test plugin installation via homebrew

/kind enhancement

Fixes #66 

cc: @csantanapr

**Release Note**

```release-note
NONE
```

**Docs**

```docs
NONE
```

### CODE REVIEW

File: .github/workflows/verify-plugins-installation.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+name: Check Knative Plugin Installation
+
+on:
+  push:
+    branches:
+      - main
```
Comment (dsimansk) [inline .github/workflows/verify-plugins-installation.yml:6]: I'd suggest a common trigger combination that we use in other Knative GH actions. 

```
on:
  push:
    branches: [ 'main', 'release-*' ]

  pull_request:
    branches: [ 'main', 'release-*' ]
```

File: .github/workflows/verify-plugins-installation.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        run: brew tap knative-sandbox/kn-plugins
+
+      - name: Install Knative plugins
+        run: |
+          brew install admin
+          brew install func
+          brew install quickstart
+          brew install source-kafka
+          brew install source-kamelet
+          brew install event
```
Comment (dsimansk) [inline .github/workflows/verify-plugins-installation.yml:34]: Can you try to extract the plugins to matrix and iterate over the items?

File: .github/workflows/verify-plugins-installation.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
+name: Check Knative Plugin Installation
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  check_plugins:
+    runs-on: ubuntu-latest
```
Comment (dsimansk) [inline .github/workflows/verify-plugins-installation.yml:10]: MacOS variant would be nice as well.

**Review by dsimansk — CHANGES_REQUESTED** [review]
Thanks for the PR! This is very much appreciated. I've added a couple of suggestions, but I looks good in general.

File: .github/workflows/verify-plugins-installation.yml (hunk 1 above)
Comment (sbdtu5498) [reply .github/workflows/verify-plugins-installation.yml:6]: Ah, I will change it accordingly.

File: .github/workflows/verify-plugins-installation.yml (hunk 2 above)
Comment (sbdtu5498) [reply .github/workflows/verify-plugins-installation.yml:34]: Yes, it would be better to keep the code dry.

**Comment by dsimansk** [comment]
@sbdtu5498 Hi there, a gentle ping! I'm wondering if you are still looking at this PR.

**Comment by sbdtu5498** [comment]
@dsimansk I will work on this soon. I was a bit busy with some other task in hand. I will try to get it merged this week positively.

**Comment by dsimansk** [comment]
> @dsimansk I will work on this soon. I was a bit busy with some other task in hand. I will try to get it merged this week positively.

Sure, no worries. Thanks for the update! It isn't super urgent to have it done this week, take your time. I wanted to confirm it's still on your radar.

**Comment by dsimansk** [comment]
Super seeded by: https://github.com/knative-extensions/homebrew-kn-plugins/pull/183

/close

**Pull request closed without merging**